parsel/selector.py

```diff
@@ -44,11 +44,14 @@ def _st(st):
         raise ValueError('Invalid type: %s' % st)
 
 
-def create_root_node(text, parser_cls, base_url=None):
+def create_root_node(text, parser_cls, base_url=None, encoding='utf8'):
     """Create root node for text using given parser class.
     """
-    body = text.strip().replace('\x00', '').encode('utf8') or b'<html/>'
-    parser = parser_cls(recover=True, encoding='utf8')
+    if isinstance(text, (six.text_type, bytes)):
+        body = text
+    elif isinstance(text, six.text_type):
+        body = text.strip().replace('\x00', '').encode(encoding) or b'<html/>'
```

Comment:
Now the `elif` will never be called.

If you want this merged before we remove Python 2 support, I think you will need to use `six.PY3` for the `if`:

```python
    if six.PY3 and isinstance(text, bytes):
        body = text
```

Regadring `strip()` and `replace()` not being used with `bytes` as input:

- Removing `strip()` _may_ be fine. It seems to be here since https://github.com/scrapy/parsel/pull/4 and I see no justification there for it. Which does not mean that it is not there for a reason, though; I just don’t know what that reason is. @kmike @eliasdorneles @dangra How’s your memory of those days? :) (I would personally not remember for sure).
- As @pawelmhm mentioned in https://github.com/scrapy/parsel/issues/210#issuecomment-781127246 , `.replace('\x00', '')` (`.replace(b'\x00', b'')` now for bytes) is needed to address #123. So I believe we need to keep this one, unless you can prove that the latest lxml version does not need this anymore, in which case we may find the lxml version that fixed it and require that as minimum version when using Parsel.

Reply:
@Gallaecio `if` condition updated.
I confirm that issue with `x00` exists so I added `replace(b'\x00', b'')`